cocos/2d/framework/ui-renderer.ts

```diff
@@ -337,6 +341,11 @@ export class UIRenderer extends Renderer {
         this.node.off(NodeEventType.ANCHOR_CHANGED, this._nodeStateChange, this);
         this.node.off(NodeEventType.SIZE_CHANGED, this._nodeStateChange, this);
         this.node.off(NodeEventType.PARENT_CHANGED, this._colorDirty, this);
+        // When disabling, it is necessary to free up idle space to fully utilize chunks
+        // and avoid breaking batch processing.
+        if (this._renderData) {
+            this.destroyRenderData();
```

Comment:
destroyRenderData() already check _renderData, so don't have check it here again.